plugins/woocommerce/tests/e2e-pw/bin/woocommerce-cleanup.php

```diff
@@ -0,0 +1,181 @@
+
+		// Set the WooCommerce currency to "United States (US) dollar"
+    update_option('woocommerce_currency', 'USD');
+
+		// Set the WooCommerce currency position to "left"
+    update_option('woocommerce_currency_pos', 'left');
+
+    // Set the WooCommerce thousand separator to a comma
+    update_option('woocommerce_price_thousand_sep', ',');
+
+    // Set the WooCommerce decimal separator to a period
+    update_option('woocommerce_price_decimal_sep', '.');
+
+    // Set the WooCommerce number of decimals to 2
+    update_option('woocommerce_price_num_decimals', '2');
+
+		// Delete all shipping zones
+    $shipping_zones = WC_Shipping_Zones::get_zones();
+    foreach ($shipping_zones as $zone) {
+        $zone_instance = new WC_Shipping_Zone($zone['zone_id']);
+        $zone_instance->delete();
+    }
+
+    // Delete the default shipping zone
+    $default_zone = new WC_Shipping_Zone(0);
+    $default_zone->delete();
+
+		// Delete all shipping classes
+    $shipping_classes = get_terms('product_shipping_class', array('hide_empty' => false));
+    foreach ($shipping_classes as $shipping_class) {
+        wp_delete_term($shipping_class->term_id, 'product_shipping_class');
+    }
+
+		// Disable all payment methods
+    $payment_gateways = WC()->payment_gateways->payment_gateways();
+    foreach ($payment_gateways as $gateway) {
+        update_option('woocommerce_' . $gateway->id . '_settings', array_merge(get_option('woocommerce_' . $gateway->id . '_settings', array()), array('enabled' => 'no')));
+    }
+
+		// Enable guest checkout
+    update_option('woocommerce_enable_guest_checkout', 'yes');
+
+		// Set the WooCommerce "From" email name
+    update_option('woocommerce_email_from_name', 'woocommerce');
+
+    // Set the WooCommerce "From" email address
+    update_option('woocommerce_email_from_address', 'wordpress@example.com');
+}
+
+add_action('admin_menu', function() {
+	add_submenu_page(
+		'woocommerce',
+		'WooCommerce Cleanup',
+		'WooCommerce Cleanup',
+		'manage_options',
+		'woocommerce-cleanup',
+		function() {
+			if (isset($_POST['wc_cleanup_reset'])) {
+				wc_cleanup_reset_site();
+				echo '<div class="updated"><p>WooCommerce site has been reset.</p></div>';
+			}
+			?>
+			<div class="wrap">
+				<h1>WooCommerce Cleanup</h1>
+				<form method="post">
+					<p>Click the button below to reset the WooCommerce site to a clean testing state.</p>
+					<p><input type="submit" name="wc_cleanup_reset" class="button button-primary" value="Reset WooCommerce Site"></p>
+				</form>
+			</div>
+			<?php
+		}
+	);
+});
+add_action('rest_api_init', function() {
+	register_rest_route('wc-cleanup/v1', '/reset', array(
+		'methods' => 'POST',
+		'callback' => 'wc_cleanup_reset_site_via_api',
+		'permission_callback' => function() {
+			$provided_key = isset($_GET['key']) ? sanitize_text_field($_GET['key']) : '';
+			$valid_key = 'FUFP2UrAbJa_.GMfs*nXne*9Fq7abvYv'; // Replace with your actual secret key
```

Comment:
Hardcoded the key (which is now published publicly).  But we can change this on every test site, or just do-away with the key altogether.